tools/splat_ext/pm_sbn.py

```diff
@@ -0,0 +1,886 @@
+        if not path.exists():
+            path.mkdir()
+        
+        with open("unknown.bin", "wb") as f:
+            f.write(self.unknown_data)
+
+        for sbn_file in self.files:
+            try:
+                filename = sbn_file.file_name()
+                sbn_file.write(path / filename)
+            except Exception as e:
+                raise Exception(f"Failed to write {sbn_file}: {e}")
+
+        with open(path / "sbn.yaml", "w") as f:
+            # Filename->ID map
+            f.write("# Mapping of filenames to entry IDs. Use 'id: auto' to automatically assign a unique ID.\n")
+            f.write("""
+# 'fakesize is an interesting case. In the final ROM, the size of a file is stored in the file header and the entry table. 
+# For bank files, however, the entry table has seemingly random data stored for file size. This data appears to be unused.
+# Rather than figure out how this data is actually generated, I hardcoded it. What does this mean for modders?
+# Well, for non bank files and bankfiles less than 65536 bytes it means nothing. Simply don't set the fakesize field and it will be fine. 
+# As for bank files that are more than 65536 bytes long, you have to set fakesize to something but it doesn't appear to matter what.\n""")
+
+            f.write("files:\n")
+            for id, sbn_file in enumerate(self.files):
+                filename = sbn_file.file_name()
+                f.write(f"  - id: 0x{id:02X}\n")
+                f.write(f"    file: {filename}\n")
+
+                if(sbn_file.fakesize != sbn_file.size):
+                    f.write(f"    fakesize: {sbn_file.fakesize}\n")
+
+            f.write("\n")
+
+            # INIT songs
+            f.write("# Songs. Use 'id: auto' to automatically assign a unique ID.\n")
+            f.write("# Symbols are used to generate the SongIDs C enum.\n")
+            f.write("songs:\n")
+            for id, entry in enumerate(self.init.song_entries):
+                bgm_file_name = self.files[entry.bgmFileIndex].file_name()
+                bk_file_names = []
+                for index in (entry.bkFile0Index, entry.bkFile1Index, entry.bkFile2Index):
+                    if index == 0:
+                        bk_file_names.append("null")
+                    else:
+                        bk_file_names.append(self.files[index].file_name())
+                f.write(f"  - id: 0x{id:02x}\n")
+                f.write(f"    symbol: {get_song_symbol_name(id)}\n")
+                f.write(f"    file: {bgm_file_name}\n")
+                f.write(f"    bk_files: [{', '.join(bk_file_names)}]\n")
+            f.write("\n")
+
+            # INIT mseqs
+            f.write("# AuGlobals::mseqFileList. Not sure why there's non-MSEQ files here!\n")
+            f.write("mseqs:\n")
+            for id, entry in enumerate(self.init.mseq_entries):
+                f.write(f"  - id: 0x{id:02x}\n")
+                f.write(f"    file: {self.files[entry].file_name()}\n")
```

Comment:
lol we should figure this out

Reply:
@HailSanta I believe it was you who named `AuGlobals::mseqFileList` - do you recall why you named it that? It seems there are non-MSEQ files in that list